### PR TITLE
040 login system

### PR DESCRIPTION
**Pull request by eoinmcbennett**

### CODE REVIEW

**Review by 40175096 — APPROVED** [review]

File: src/main/java/org/kainos/ea/team2/api/AuthenticationService.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+        if (!Arrays.equals(credentialHash, passwordHash)) {
+            return null;
+        }
+
+        return new JWT()
+            .setIssuer("org.kainos.ea")
+            .setIssuedAt(ZonedDateTime.now(ZoneOffset.UTC))
+            .setSubject(credentials.getUsername())
+            .setExpiration(ZonedDateTime.now(ZoneOffset.UTC).plusDays(1));
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/team2/api/AuthenticationService.java:148]: you could also include the role in the token so you wouldn't need to check the db

File: sqlfiles/002-users-table.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+USE MovesLikeSwagger_MeganM;
+
+CREATE TABLE `Users` (
```
Comment (shaunganley) [inline sqlfiles/002-users-table.sql:3]: should a user also have a role?

File: src/test/java/org/kainos/ea/team2/integration/AuthIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            MovesLikeSwaggerApplication.class, null,
+            new ResourceConfigurationSourceProvider()
+    );
+
+    /**
+     * Verify that logging in valid credentials gives back a valid JWT
+     */
+    @Test
+    void login_shouldReturnValidJWT_whenValidCredentialsArePassed() {
+        BasicCredentials credentials = new BasicCredentials("test@kainos.com","testing");
```
Comment (shaunganley) [inline src/test/java/org/kainos/ea/team2/integration/AuthIntegrationTest.java]: you probably don't want to hardcode valid credentials, could you pull these in from env vars?

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: src/test/java/org/kainos/ea/team2/integration/AuthIntegrationTest.java (hunk 3 above)
Comment (eoinmcbennett) [reply src/test/java/org/kainos/ea/team2/integration/AuthIntegrationTest.java]: Changed to pull from env vars.

File: src/test/java/org/kainos/ea/team2/integration/AuthIntegrationTest.java [hunk 4]
```diff
@@ -23,12 +23,19 @@ public class AuthIntegrationTest {
...
      * Verify that logging in valid credentials gives back a valid JWT
      */
     @Test
     void login_shouldReturnValidJWT_whenValidCredentialsArePassed() {
-        BasicCredentials credentials = new BasicCredentials("test@kainos.com","testing");
+        if(VALID_USER_NAME == null || VALID_USER_PASSWORD == null){
+            throw new IllegalArgumentException("Test credential environment variables not set!");
+        }
+        BasicCredentials credentials = new BasicCredentials(VALID_USER_NAME,VALID_USER_PASSWORD);
+        System.out.println(credentials.getUsername());
```
Comment (shaunganley) [inline src/test/java/org/kainos/ea/team2/integration/AuthIntegrationTest.java]: you can remove this

**Review by shaunganley — APPROVED** [review]

**Pull request merged**